### ISSUE TITLE
⚡ Interactions/Animations engine

### ISSUE THREAD

**Issue by itsNintu** [webstudio-is/webstudio#2645]
The upcoming animations engine will allows users to easily implement complex interactions and animations, enhancing the site experience and engagement. From subtle hover effects to intricate motion graphics.

**Comment by GGIdols** [comment]
looking forward for some animations

**Comment by psdcovers** [comment]
Will this include interacting with spline.design content via their simple api?

**Comment by CarloBu** [comment]
I really can't wait for the interactions/animation support. It's great that I can use GSAP of course, but it's time-consuming and breaks the visual design flow and speed. 
Can I ask if the webstudio planning to use framer motion or motion one under the hood? 
Framer Motion looks like a perfect fit for react based project and it has an MIT license. 
Designers are jumping to Webflow/FramerWix because they can let designers create AAA-level interactions/animations without much effort and quickly.

**Comment by kof** [comment]
@CarloBu we haven't decided on specific implementation details, we do have lots of ideas.

**Comment by CarloBu** [comment]
@kof 
Thank you for your quick answer. 
I just found this project, and I was very excited to see that I can copy and paste the entire Relume template. I don't lock myself or my clients into closed-source corporations, but Webflow + Relume is too good to dismiss. I was experimenting with live conversion from Webflow to HTML code to speed up website development using Relume templating when I found Webstudio. This is my dream come true! Thank you!

I understand that there is a lot on your plate right now and that animations are not a priority.

Right now Webflow and Framer have an edge in the web builder space because of amazing animations. Many Awwwards websites are built on these platforms, and I believe when the Webstudio team introduces the drag-and-drop animation feature - Webstudio will explode within the designer community.

**Comment by kof** [comment]
@CarloBu copy paste from webflow isn't yet officially launched, we have a bunch of bugs to fix. Please don't post about it on  social networks yet)

**Comment by CarloBu** [comment]
Yeah, I noticed, that this is experimental at the moment :) A lot of components are still not supported, like instance components that do not convert to slots, but the Relume collection is basically working (except interactions), which is what I wanted.

That was my entire dream: generate a website wireframe in minutes, migrate the website to the web builder, apply the style, add the CMS, and ship it with a fraction of the effort compared to the current Astro + Tailwind workflow(this is already fast). Relume + Webstudio = dream combo :)

**Comment by kof** [comment]
I am afraid interactions are not going to be supported for copy-paste, as well as most complex components from advanced and media sections.

**Comment by kof** [comment]
Relume and similar libraries are indeed our focus.

**Comment by chirantanaditya** [comment]
Does anyone know how we can add lottie animation in Web Studio?

**Comment by kof** [comment]
> Does anyone know how we can add lottie animation in Web Studio?

just like any external widget using html embed and their embed code, did you try? did it not work?

**Comment by chirantanaditya** [comment]
It would be great if you could integrate Motion.dev

**Comment by TrySound** [comment]
@chirantanaditya You can integrate it too with html embed

**Comment by chirantanaditya** [comment]
> [@chirantanaditya](https://github.com/chirantanaditya) You can integrate it too with html embed

That's great but would love to have it inbuilt. Also, I'll look around for tutorials for this. Any recommendations?

**Comment by kof** [comment]
Component based approach to animations


## View Component

We add a special component "View". User adds it on canvas as usual and puts everything inside that will be animated when instance enters the viewport.

1.  Prop "Start Animation" with a select box for values like bounce, flash, etc https://animate.style/ and the same thing but for "End Animation"
1. If a complex animation is selected, we show additional props that can be chosen to customize it, eg.. fly = (x, y, opacity, easing)a
1. If user didn't choose a complex animation like bounce, they can choose a simple easing function e.g. ease-in, ease-in-cubic.
1.  Prop "axis" https://developer.mozilla.org/en-US/docs/Web/CSS/animation-timeline/view
1. You can nest View inside a View 
1. Prop `trigger: 'self' | 'nearest' = 'self'` allows you to specify the event emitter, self by default
1. Prop `duration`  
1. Child component animation will be composed with parent component animation

## Scroller Component

Similar to view but triggers animation on scroll.

1. Prop `scroller: 'self' | 'nearest' | 'root' = 'self'` allows you to specify the event emitter, self by default
1. All the same props as View Component

**Comment by TrySound** [comment]
Let's rename View to Reveal or Entry to not confuse in the future. `view()` is scroll-driven animation and not single event, its timeline starts when element entry into viewport and ends when it leaves viewport.

**Comment by kof** [comment]
Basically the difference between view() and reveal is that reveal does not have exit, it only has entry

**Comment by TrySound** [comment]
The difference is that view is controlled by scroll and reveal is controlled by time and only triggered by scroll.
